src/library.js

```diff
@@ -3506,6 +3506,10 @@ mergeInto(LibraryManager.library, {
       return;
     }
     try {
+#if ASSERTIONS
+      assert(!inUserCode, 'callUserCallback called when already running user code');
+      inUserCode = true;
+#endif
```

Comment:
Why is this an error?

Reply:
If we did have such uses, then they would show up as test failures here, right?

If you are already in a user stack then you can just invoke the callback yourself.. what would be the point of using `callUserCallback` in that case?  You wouldn't want it to do any of the things it normally does... which is specifically why @dschuff added the, now-removed, `syncronous` argument.. which bypassed what `calllUserCallback` was actually doing.

Anyone assuming they can use `callUserCallback` when they are already in a user stack would already be broken, as in this `glfw` bug.

Reply:
The first live of the docs seem pretty clear to me: `Used to call user callbacks from the embedder / event loop.`

If you are not coming from the event loop you simply don't need to call this function.

This new assertion just helps such users know that they are making a mistake.

Reply:
How about if I made this into a warnOnce rather than an assertion?  I think that might be a better way to go.